src/components/tests/settings-drawer.test.tsx

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import { createMount } from '@material-ui/core/test-utils';
+
+import SettingsDrawer from '../settings-drawer';
+import {
+  loadSettings,
+  TrajectoryAnimation,
+  Settings,
+  TrajectoryDiameter,
+  TrajectoryColor,
+} from '../../settings';
+import { SettingsContext } from '../app-contexts';
+
+const mount = createMount();
+
+describe('Settings Drawer', () => {
+  let settings = loadSettings();
+  let onSettingsChange: jest.Mock;
+
+  beforeEach(() => {
+    onSettingsChange = jest.fn();
+  });
+
+  it('should render correctly', () => {
+    const wrapper = mount(
+      <SettingsContext.Provider value={settings}>
+        <SettingsDrawer settings={settings} onSettingsChange={onSettingsChange} open={true} />
+      </SettingsContext.Provider>,
+    );
+    wrapper.unmount();
+  });
+
+  it('should call onSettingsChange function when handleTrajectoryAnimationChange is called', () => {
+    const newSettings: Settings = {
+      ...settings,
+      trajectoryAnimation: TrajectoryAnimation.Outline,
+    };
+    const mockEvent = { target: { value: TrajectoryAnimation.Outline } };
+    const wrapper = mount(
+      <SettingsContext.Provider value={newSettings}>
+        <SettingsDrawer
+          settings={settings}
+          onSettingsChange={newSettings => onSettingsChange(newSettings)}
+          open={true}
+        />
+      </SettingsContext.Provider>,
+    );
+
+    wrapper
+      .find('input')
+      .at(3)
+      .simulate('change', mockEvent);
+    expect(onSettingsChange).toHaveBeenCalledTimes(1);
+  });
+
+  it('should call onSettingsChange function when handleTrajectoryDiameterChange is called', () => {
+    const newSettings: Settings = {
+      ...settings,
+      trajectoryDiameter: TrajectoryDiameter.FixSize,
+    };
+    const mockEvent = { target: { value: TrajectoryDiameter.FixSize } };
+    const wrapper = mount(
+      <SettingsContext.Provider value={newSettings}>
+        <SettingsDrawer
+          settings={settings}
+          onSettingsChange={newSettings => onSettingsChange(newSettings)}
+          open={true}
+        />
+      </SettingsContext.Provider>,
+    );
+
+    wrapper
+      .find('input')
+      .at(4)
+      .simulate('change', mockEvent);
+    expect(onSettingsChange).toHaveBeenCalledTimes(1);
+  });
+
+  it('should call onSettingsChange function when handleTrajectoryColorChange is called', () => {
+    const newSettings: Settings = {
+      ...settings,
+      trajectoryColor: TrajectoryColor.Theme,
+    };
+    const mockEvent = { target: { value: TrajectoryColor.Theme } };
+    const wrapper = mount(
+      <SettingsContext.Provider value={newSettings}>
+        <SettingsDrawer
+          settings={settings}
+          onSettingsChange={newSettings => onSettingsChange(newSettings)}
+          open={true}
+        />
+      </SettingsContext.Provider>,
+    );
+
+    wrapper
+      .find('input')
+      .at(6)
```

Comment:
What I mean was that you can add something like this in the setting-drawer:

```tsx
  {trajColorText.map((text, i) => (
            <FormControlLabel
              key={i}
              className={classes.flexBasis}
              value={i}
              control={<Radio name={text} />}
              label={text}
            />
          ))} 
```

And find it this way on your tests:

```
    wrapper.find('[name="Theme"]').find('input').simulate('change', mockEvent);
    expect(onSettingsChange).toHaveBeenCalledTimes(1);
```

Because if you are using at(x) you are depending on the position of the inputs, which is not good because if in the future we change the positions of those (radio) inputs we will have to modify our tests